common/src/main/java/earth/terrarium/adastra/api/events/AdAstraEvents.java

```diff
@@ -43,6 +47,26 @@ static boolean post(ServerLevel level, LivingEntity entity) {
         }
     }
 
+    @FunctionalInterface
+    public interface EntityHasOxygenEvent {
+        Optional<Boolean> tick(Entity entity, boolean hasOxygen);
+
+        static void register(EntityHasOxygenEvent listener) {
+            ENTITY_HAS_OXYGEN_LISTENERS.add(listener);
+        }
+
+        @ApiStatus.Internal
+        static boolean post(Entity entity, boolean hasOxygen) {
+            for (var listener : ENTITY_HAS_OXYGEN_LISTENERS) {
+                Optional<Boolean> newHasOxygen = listener.tick(entity, hasOxygen);
```

Comment:
`newHasOxygen` sounds very weird in English, should just be called `newOxygen`

Reply:
changed